python/pynessie/client/nessie_client.py

```diff
@@ -259,7 +261,7 @@ def merge(self, from_ref: str, onto_branch: str, from_hash: Optional[str] = None
             from_hash = from_hash_ref
 
         merge_json = MergeSchema().dump(Merge(from_ref, str(from_hash)))
-        merge(self._base_url, self._auth, onto_branch, merge_json, old_hash, self._ssl_verify)
+        return MergeResponseSchema().load(merge(self._base_url, self._auth, onto_branch, merge_json, old_hash, self._ssl_verify))
```

Comment:
nit: same as above

@snazy wdyt is our preferred strategy in terms of compatibility of nessie CLI against different nessie versions?
i.e. do we need to gracefully handle getting an empty response here?

Reply:
Nessie CLI should also work against older versions, if possible.

Reply:
_endpoints passes an empty dict as response for old servers that don't return any response on merge request. This means normal output will remain same and --json output will be a json with each field null.
`{"targetCommits": null, "sourceCommits": null, "targetBranch": null, "details": null, "wasApplied": false, "expectedHash": null, "wasSuccessful": false, "resultantTargetHash": null, "commonAncestor": null, "effectiveTargetHash": null}`

Reply:
Updated message for legacy servers